egs/tedlium3/ASR/local/convert_transcript_words_to_bpe_ids.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Xiaomi Corporation (Author: Mingshuang Luo)
+"""
+Convert a transcript based on words to a list of BPE ids.
+
+For example, if we use 2 as the encoding id of <unk>:
+
+texts = ['this is a <unk> day']
+spm_ids = [[38, 33, 6, 2, 316]]
+
+texts = ['<unk> this is a sunny day']
+spm_ids = [[2, 38, 33, 6, 118, 11, 11, 21, 316]]
+
+texts = ['<unk>']
+spm_ids = [[2]]
+"""
+
+import argparse
+import logging
+from typing import List
+
+import sentencepiece as spm
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--texts", type=List[str], help="The input transcripts list."
+    )
+    parser.add_argument(
+        "--unk-id",
+        type=int,
+        default=2,
+        help="The number id for the token '<unk>'.",
+    )
```

Comment:
```suggestion
```

We can get the id for `<unk>` directly from the BPE model, so there is no need to pass it here.